contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -742,6 +744,19 @@ tsql_has_func_args_higher_precedence(int n, Oid *inputtypes, FuncCandidateList c
 	if (is_vectorized_binary_operator(candidate1) && !is_vectorized_binary_operator(candidate2))
 		return true;
 
+	/* Prioritize candidates with same-typed arguments for operators only*/
+	if (candidates_are_opers)
+	{
+		for (i = 1; i < n; ++i)
+		{
+			can1_is_sametype &= argtypes1[i-1] == argtypes1[i];
+			can2_is_sametype &= argtypes2[i-1] == argtypes2[i];
+		}
```

Comment:
there could be unary/prefix operators possible, should we handle it?

Reply:
If we have a unary operator, then n should be 1, meaning we won't enter the for loop and can1/2_is_sametype are both true.